### PR TITLE
Adds an age field to IDs

### PR DESCRIPTION
**Pull request by Pastamadness**
## What this does
Adds an AGE field to IDs that can be set by the HoP or when imprinting biometric data at a medical scanner (assume that a company that has mastered cloning has also mastered measuring the age of a subject).
Agent IDs can edit this field, too.

Admittedly, I have no idea what I was doing with the href list or NanoUI. Things appear work when tested but I'd appreciate some feedback on those copy-pasted parts of this PR.

## Why it's good

Adds minute flavor and enables new RP interactions such as ageism or refusing to serve drinks as a bartender when your client has not reached the arbitrary legal drinking age you've chosen.
## Changelog

:cl:
 * rscadd: IDs now contain an age field.

### CODE REVIEW

File: code/game/machinery/computer/card.dm [hunk 1]
```diff
@@ -334,6 +335,18 @@
 					modify.update_virtual_wallet()
 			nanomanager.update_uis(src)
 
+		if ("age")
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:338]: invert check to avoid rightward creep

File: code/game/machinery/computer/card.dm [hunk 2]
```diff
@@ -334,6 +335,18 @@
...
+			if (is_authenticated())
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:339]: invert check to avoid rightward creep

File: code/game/machinery/computer/card.dm [hunk 3]
```diff
@@ -334,6 +335,18 @@
...
+				var/t2 = modify
+				if ((modify == t2 && (in_range(src, usr) || (istype(usr, /mob/living/silicon))) && istype(loc, /turf)))
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:341]: probably better if the machine just handled sanity at the top

File: code/game/machinery/computer/card.dm [hunk 4]
```diff
@@ -161,6 +161,7 @@
 	data["scan_name"] = scan ? scan.name : "-----"
 	data["authenticated"] = is_authenticated()
 	data["has_modify"] = !!modify
+	data["age"] = modify ? modify.age_id : "-----"
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:164]: this many dashes seems odd given that it should be 2 digits

File: code/game/machinery/computer/card.dm [hunk 5]
```diff
@@ -334,6 +335,18 @@
...
+		if ("age")
+			if (is_authenticated())
+				var/t2 = modify
+				if ((modify == t2 && (in_range(src, usr) || (istype(usr, /mob/living/silicon))) && istype(loc, /turf)))
+					var/temp_age = text2num(href_list["age"])
+					if(temp_age)
+						modify.age_id = trunc(abs((temp_age)))
+					else
+						src.visible_message("<span class='notice'>[src] buzzes rudely.</span>")
+					modify.update_virtual_wallet()
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:347]: what relationship would this have with the wallet?

File: code/game/machinery/computer/card.dm [hunk 6]
```diff
@@ -334,6 +335,18 @@
 					modify.update_virtual_wallet()
 			nanomanager.update_uis(src)
 
+		if ("age")
+			if (is_authenticated())
+				var/t2 = modify
+				if ((modify == t2 && (in_range(src, usr) || (istype(usr, /mob/living/silicon))) && istype(loc, /turf)))
+					var/temp_age = text2num(href_list["age"])
+					if(temp_age)
+						modify.age_id = trunc(abs((temp_age)))
```
Comment (Kurfursten) [inline code/game/machinery/computer/card.dm:344]: still permits very large numbers that can cause errors, like for example 999999999999999999999999999999999999999999999999999999999999999999999

just clamp it

File: code/game/machinery/computer/card.dm (hunk 4 above)
Comment (Eneocho) [reply code/game/machinery/computer/card.dm:164]: there goes the antediluvian RP

**Comment by Pastamadness** [comment]
Being extremely lazy about this but I promise to fix stuff and reply to feedback this weekend

**Comment by west3436** [comment]
@Pastamadness hello are you going to work on this again?

**Comment by Pastamadness** [comment]
> @Pastamadness hello are you going to work on this again?

Yeah it's only been 2 days since I've opened this PR, rig-

**Review by jwhitak — CHANGES_REQUESTED** [review]
As above, gonna mark this as requested changes until it's ready.

**Comment by Pastamadness** [comment]
I've been hit a severe case of ennui once I lost focus on this. I'll touch this again later.

**Pull request closed without merging**